### PR TITLE
CORTX-32074 1N Deployment job in PR check for cortx-rgw-integration, cortx-rgw

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- Enable PR check job for the RGW and RGW-integration component. The job is build in 5 stages Build the RPM, release It and the create image out of the rmp and taking rest of the component rpm as latest and create their images. use the images to 1N deployment. These Jobs going to run on each PR for the RGW and RGW-integration branch and validate the code by deploying a 1N cluster. 

# Design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing
   -  Jenkins Job for rgw-integration : https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/rgw-integration/12/
   - Jenkins Job for cortx-rgw : https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/cortx-rgw/10/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        timeout(time: 180, unit: 'MINUTES')
+        timestamps()
+        disableConcurrentBuilds()
+        ansiColor('xterm')  
+    }
+
+    parameters {  
+	    string(name: 'RGW_URL', defaultValue: 'https://github.com/Seagate/cortx-rgw', description: 'Repo for rgw')
+        string(name: 'RGW_BRANCH', defaultValue: 'main', description: 'Branch for rgw')
+        choice(name: 'DEPLOY_BUILD_ON_NODES', choices: ["Both", "1node", "3node" ], description: '''<pre>If you select Both then build will be deploy on 1 node as well as 3 node. If you select 1 node then build will be deploy on 1 node only. If you select 3 node then build will be deploy on 3 node only.
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:19]: Please use `1node` as default

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    parameters {  
+	    string(name: 'RGW_URL', defaultValue: 'https://github.com/Seagate/cortx-rgw', description: 'Repo for rgw')
+        string(name: 'RGW_BRANCH', defaultValue: 'main', description: 'Branch for rgw')
+        choice(name: 'DEPLOY_BUILD_ON_NODES', choices: ["Both", "1node", "3node" ], description: '''<pre>If you select Both then build will be deploy on 1 node as well as 3 node. If you select 1 node then build will be deploy on 1 node only. If you select 3 node then build will be deploy on 3 node only. 
+</pre>''')
+        string(name: 'CORTX_RE_URL', defaultValue: 'https://github.com/Seagate/cortx-re', description: 'Repo for cortx-re')
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch for cortx-re')
+
+        choice (
+            choices: ['all', 'cortx-all', 'cortx-control', 'cortx-rgw', 'cortx-data'],
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:25]: Remove `cortx-all` image

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        GPR_REPO = "https://github.com/${ghprbGhRepository}"
+        RGW_URL = "${ghprbGhRepository != null ? GPR_REPO : RGW_URL}"
+        RGW_BRANCH = "${sha1 != null ? sha1 : RGW_BRANCH}"
+
+        RGW_GPR_REFSEPEC = "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
+        RGW_BRANCH_REFSEPEC = "+refs/heads/*:refs/remotes/origin/*"
+        RGW_PR_REFSEPEC = "${ghprbPullId != null ? RGW_GPR_REFSEPEC : RGW_BRANCH_REFSEPEC}"
+
+        //////////////////////////////// BUILD VARS //////////////////////////////////////////////////
+        // OS_VERSION, singlenode_host, threenode_hosts, COMPONENTS_BRANCH and CORTX_SCRIPTS_BRANCH are manually created parameters in jenkins job.
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:44]: Do not take `CORTX_SCRIPTS_BRANCH` as input

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please fill all the details in PR template
Please add Jenkins job used for validation

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:25]: Ok

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 1 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy]: Done

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:20]: Done.

File: jenkins/internal-ci/generic/pr/cortx_deployment/rgw-integration.groovy [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+                        done
+                    popd
+                    # Symlink 3rdparty repo artifacts
+                    ln -s "${THIRD_PARTY_DEPS}" "${THIRD_PARTY_LOCATION}"
+                        
+                    # Symlink python dependencies
+                    ln -s "${PYTHON_DEPS}" "${PYTHON_LIB_LOCATION}"
+                '''
+
+                // sh label: 'RPM Signing', script: '''
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/rgw-integration.groovy:156]: Remove `rpm signing` block. We do not need it for internal tests

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+                                cp $component/last_successful/*.rpm "${CORTX_ISO_LOCATION}"
+                            fi
+                        done
+                    popd
+                    # Symlink 3rdparty repo artifacts
+                    ln -s "${THIRD_PARTY_DEPS}" "${THIRD_PARTY_LOCATION}"
+                    # Symlink python dependencies
+                    ln -s "${PYTHON_DEPS}" "${PYTHON_LIB_LOCATION}"
+                '''
+                // sh label: 'RPM Signing', script: '''
```
Comment (shailesh-vaidya) [inline jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:146]: Remove `rpm signing` block. We do not need it for internal tests

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/internal-ci/generic/pr/cortx_deployment/rgw-integration.groovy (hunk 4 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/rgw-integration.groovy]: Done.

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 5 above)
Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:146]: Ok.sure

Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy]: done

File: jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy (hunk 2 above)
Comment (shailesh-vaidya) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:20]: @gauravchaudhari02  - Can you please check this. Ideally, we should not be using `cortx-all` images right

Comment (Abhirup-A-G) [reply jenkins/internal-ci/generic/pr/cortx_deployment/cortx-rgw.groovy:20]: Removed it

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please rename rgw-integration.groovy to cortx-rgw-integration.groovy . To maintain same nomenclature as [cortx-rgw-integration.groovy](https://github.com/Abhirup-A-G/cortx-re/blob/CORTX-32074/jenkins/internal-ci/generic/branch/cortx-rgw-integration.groovy)

**Review by shailesh-vaidya — COMMENTED** [review]
LGTM 
Please test once with providing feature branch in https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/Generic/job/cortx-rgw/10/parameters/

**Review by shailesh-vaidya — APPROVED** [review]

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**